### PR TITLE
make it compatible with also odd channels

### PR DESCRIPTION
**Pull request by MiladInk**
Hi Lucas,

I wanted to make the code compatible with also odd number of channels for additive and affine coupling. It was fairlt straightforward and I think it is good to have it. It runs fine in my computer right now. Please tell me how you think about it or if you seen any shortcoming.

### CODE REVIEW

File: invertible_layers.py [hunk 1]
```diff
@@ -234,8 +234,8 @@ def reverse_(self, x, objective):
 class AdditiveCoupling(Layer):
     def __init__(self, num_features):
         super(AdditiveCoupling, self).__init__()
-        assert num_features % 2 == 0
-        self.NN = NN(num_features // 2)
+        # assert num_features % 2 == 0
+        self.NN = NN(math.ceil(num_features / 2), channels_out=math.floor(num_features / 2) * 2)
```
Comment (pclucas14) [inline invertible_layers.py]: hey, looking at the old code it seems that `self.NN` took `num_features // 2` channels in and returned `num_features // 2` channels out. However it the new code the number of output channels seems to be doubled?

Comment (MiladInk) [reply invertible_layers.py]: Ah, sure, this problem is for the additive coupling layer. I will correct it. The affine looks fine yes?

Comment (MiladInk) [reply invertible_layers.py]: I have changed the code for additive. Does it look fine now?

Comment (pclucas14) [reply invertible_layers.py]: looks good. I'm assuming you checked that it works now for both odd and even n_channels ?

**Comment by MiladInk** [comment]
Hi Lucsa, I have resolved the issue with additive coupling layer with doubled channels. Would you mind taking a look and tell me what you think? Thank you very much.

**Comment by pclucas14** [comment]
looks good. I'm assuming you checked that it works now for both odd and even n_channels ?

File: invertible_layers.py (hunk 1 above)
Comment (MiladInk) [reply invertible_layers.py]: I am using it in my own work right now and it works. but I have changed a few things in my code too. So please give me a little time to also test it on your own code without my changes and I will leave another comment as soon as I did it. So, I think do not accept the pull request right now so I can be 100% sure.

**Comment by MiladInk** [comment]
I am using it in my own work right now and it works. but I have changed a few things in my code too. So please give me a little time to also test it on your own code without my changes(I mean all my other changes to the code expect this one) and I will leave another comment as soon as I did it. So, I think do not accept the pull request right now so I can be 100% sure and then I will tell you here asap I tested it.

File: invertible_layers.py (hunk 1 above)
Comment (pclucas14) [reply invertible_layers.py]: sounds good, there's no rush. thanks for the PR!

**Pull request closed without merging**